src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1047,13 +1047,40 @@ private boolean addEntry(final Map<String, List<MapEntry>> entryMap, final Strin
         return map;
     }
 
-	/**
+    private boolean isValidAliasPath(final String path){
+            if (path == null) {
+                throw new IllegalArgumentException("Unexpected null path");
+            }
+
+          // ignore system tree
+          if (path.startsWith(JCR_SYSTEM_PREFIX)) {
+            log.debug("loadAliases: Ignoring {}", path);
+            return false;
+          }
+
+            // check white list
+            if ( this.factory.getAliasPath() != null && !this.factory.getAliasPath().isEmpty()) {
+                boolean allowed = false;
+                for(final String prefix : this.factory.getAliasPath()) {
```

Comment:
Nit: this could be much nicer with streams:

`boolean allowed = this.factory.getAliasPath().stream().anyMatch(path::startsWith)`

(Disclaimer: the above was typed in this comment box, so most likely an error is hiding somewhere)